IO/IO_Ex1_Test.cpp

```diff
@@ -0,0 +1,169 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <stdexcept>
+
+#include "Ex1.hpp"
+
+void simpleCopyTest();
+void simpleMoveTest();
+void simpleMoveOpTest();
+
+void checkFileLineCount(const std::string &file_name, int expected_lines);
+void createFile(const std::string filename);
+
+int main() {
+
+    try {
+        simpleCopyTest();
+        simpleMoveTest();
+        simpleMoveOpTest();
+
+        std::cout << "All tests passed successfully!" << std::endl;
+    } catch (const std::exception &e) {
+        std::cerr << "Test failed: " << e.what() << std::endl;
+    }
+
+    return 0;
+}
+
+void simpleCopyTest() {
+
+    const std::string file_name = "test_file1.txt";
+    std::ios_base::openmode prem = std::ios::in | std::ios::out;
+
+    createFile(file_name);
+
+    FileHandler file1(file_name, prem);
+
+    std::fstream &file1_fd = file1.getFile();
```

Comment:
Didn't you removed the getFile() method?
Please fix your tests so they will compile.

Reply:
i removed it and the old tests